builtin/providers/azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -297,9 +305,21 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 							Set:      schema.HashString,
 						},
 
+						"managed_disk_type": {
+							Type:          schema.TypeString,
+							Optional:      true,
+							Computed:      true,
+							ConflictsWith: []string{"storage_profile_os_disk.vhd_containers"},
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.PremiumLRS),
+								string(compute.StandardLRS),
+							}, true),
+						},
+
 						"caching": {
 							Type:     schema.TypeString,
-							Required: true,
+							Optional: true,
+							Computed: true,
```

Comment:
Thank you for reviewing @tombuildsstuff! :) This is the same implementation as `virtual_machines` so wanted to keep it consistent with that behavior and documented default in documentation now. Please let me know if this works or if I should add default value here.